deploy/crds/kieapp.crd.yaml

```diff
@@ -97,45 +93,261 @@ spec:
                 server:
```

Comment:
We have at least a naming problem here. Maybe can be solved by bringing "servers" under this "server"... maybe calling it something else, or maybe leave as servers

Reply:
I decided to let the consumer decide whether to use a common `server` configuration or a specific one for each server.
```
server:
  deployments: 2
```
or
```
servers:
 - name: server1
 - name: server2
```

Reply:
what if both are set? does it default to one or the other

Reply:
It will fail

Reply:
Let’s add some logic to prevent that